src/client/components/2dContainer/blueprint.controller.js

```diff
@@ -20,38 +21,492 @@ export default class BlueprintEditor {
+                points: [
+                    {}, {},
+                ],
+            },
+
+            frame13: {
+                count: 13,
+                empty: true,
+                line: true,
+                color: 'red',
+                width: 2,
+                text: false,
+                size: 0,
+                points: [
+                    {}, {}, {}, {}, {},
+                ],
+            },
+
+            frame13Side: {
+                count: 13,
+                empty: true,
+                line: true,
+                color: 'invisible',
+                width: 2,
+                text: false,
+                size: 0,
+                variable: 'frameSide',
+                points: [
+                    {}, {},
+                ],
+            },
+
+            frame14: {
+                count: 14,
+                empty: true,
+                line: true,
+                color: 'red',
+                width: 2,
+                text: false,
+                size: 0,
+                points: [
+                    {}, {}, {}, {}, {},
+                ],
+            },
+
+            frame14Side: {
+                count: 14,
+                empty: true,
+                line: true,
+                color: 'invisible',
+                width: 2,
+                text: false,
+                size: 0,
+                variable: 'frameSide',
+                points: [
+                    {}, {},
+                ],
+            },
+
+            frame15: {
+                count: 15,
+                empty: true,
+                line: true,
+                color: 'red',
+                width: 2,
+                text: false,
+                size: 0,
+                points: [
+                    {}, {}, {}, {}, {},
+                ],
+            },
+
+            frame15Side: {
+                count: 15,
+                empty: true,
+                line: true,
+                color: 'invisible',
+                width: 2,
+                text: false,
+                size: 0,
+                variable: 'frameSide',
+                points: [
+                    {}, {},
+                ],
+            },
         };
-        Object.keys(Maths).forEach((key) => {
-            Object.keys(copy).forEach((key1) => {
-                if (Math.abs(Maths[key] - copy[key1]) > maxDiff) {
-                    maxDiff = Math.abs(Maths[key] - copy[key1]);
-                    console.log(maxDiff);
-                    if (Maths[key] < Maths[key1]) {
-                        vals.minVal = key;
-                        vals.maxVal = key1;
+
+        // Find the location of each frame and insert their offsets into the structure
+        let count = 0;
+        let startY;
+        boat.frames.forEach((frame, index) => {
+            const {locationA, locationB, locationC} = findLocation(boat, frame);
+            location1 = locationA;
+            location2 = locationB;
+            location3 = locationC;
```

Comment:
Since this is the only place that you use these location variables, you can actually just declare them within the loop. Due to the nature of Javascripts garbage collection mechanism, at the end of each iteration, the memory will be released automatically.